centreon/www/include/configuration/configObject/service/DB-Func.php

```diff
@@ -2919,23 +2919,22 @@ function updateServiceNotifOptionInterval_MC($service_id = null)
 
 function updateServiceNotifOptionTimeperiod($service_id = null, $ret = array())
 {
-    global $form, $pearDB;
+    global $pearDB;
 
-    if (!$service_id) {
+    if (! $service_id) {
         return;
     }
 
-    if (isset($ret["timeperiod_tp_id2"])) {
-        $ret = $ret["timeperiod_tp_id2"];
-    } else {
-        $ret = $form->getSubmitValue("timeperiod_tp_id2");
-    }
-
-    $rq = "UPDATE service SET ";
-    $rq .= "timeperiod_tp_id2 = ";
-    isset($ret) && $ret != null ? $rq .= "'" . $ret . "' " : $rq .= "NULL ";
-    $rq .= "WHERE service_id = '" . $service_id . "'";
-    $dbResult = $pearDB->query($rq);
+    $request = <<<'SQL'
+        UPDATE `service` SET `timeperiod_tp_id2` = :timeperiod_tp_id2
+        WHERE `service_id` = :service_id
+        SQL;
+    $statement = $pearDB->prepare($request);
+    isset($ret['timeperiod_tp_id2'])
+        ? $statement->bindValue(':timeperiod_tp_id2', $ret['timeperiod_tp_id2'], \PDO::PARAM_STR)
+        : $statement->bindValue(':timeperiod_tp_id2', null, \PDO::PARAM_NULL);
+    $statement->bindValue(':service_id', $service_id, \PDO::PARAM_INT);
+    $statement->execute();
```

Comment:
Now, it's recommended to use the new methods of CentreonDB : prepareQuery(), executePreparedQuery(). The bindValue method is not necessary here.

Reply:
It seems that using these methods in such a simple query is an additional level of abstraction

Reply:
These methods have been added to have an abstraction to have a better error handling (logs etc) and to have only one exception to catch : CentreonDBException.  It's the purpose of this connector and these new methods as an adapter if you want ;)

Reply:
Done

Reply:
Thanks for your change but don't forget to handle the CentreonDBException. You can catch it and return null with a log of CentreonLog :)

Reply:
Done